packages/mui-material/src/IconButton/IconButton.js

```diff
@@ -215,9 +216,14 @@ const IconButton = React.forwardRef(function IconButton(inProps, ref) {
       {...other}
       ownerState={ownerState}
     >
-      <IconButtonLoadingIndicator className={classes.loadingIndicator} ownerState={ownerState}>
-        {loading && loadingIndicator}
-      </IconButtonLoadingIndicator>
+      {typeof loading === 'boolean' && (
+        // use plain HTML span to minimize the runtime overhead
+        <span className={classes.loadingWrapper} style={{ display: 'contents' }}>
+          <IconButtonLoadingIndicator className={classes.loadingIndicator} ownerState={ownerState}>
+            {loading && loadingIndicator}
+          </IconButtonLoadingIndicator>
+        </span>
+      )}
```

Comment:
Do the same as [Button](https://github.com/mui/material-ui/blob/master/packages/mui-material/src/Button/Button.js#L581-L591)